pages/_conferenceCode/contacts/index.vue

```diff
@@ -338,56 +352,128 @@ function initData (){
+// ===================
+// Mounted method
+// ===================
+async function mounted (){
+  const qs = this.$route.query
+
+  this.meetingsOptions = await getMajorMinorMeetings.apply(this)
+  this.countryList = this.$store.getters['countries/asSelectOptions']
+  this.countryList = _.sortBy(this.countryList, [ o => o.Name ])
+  this.countryOptions = this.countryList
+  this.organizationTypesOptions = this.$store.getters[
+    'organizations/asSelectOptions'
+  ]
+
+  await setQueryStringFilters.call(this, qs)
+}
+
+function isFiltersApplied (){
   if (
-    _this.filter.name ||
-    _this.filter.isBroadSearch ||
-    _this.filter.selectedMeetings.length ||
-    _this.filter.selectedOrganizationType ||
-    _this.filter.selectedOrganization.length ||
-    _this.filter.selectedCountry.length ||
-    _this.filter.selectedFlag.length ||
-    _this.filter.selectedAttendance.length ||
-    _this.filter.selectedScop
+    this.filter.name ||
+    this.filter.isBroadSearch ||
+    this.filter.selectedMeetings.length ||
+    this.filter.selectedOrganizationType ||
+    this.filter.selectedOrganizations.length ||
+    this.filter.selectedCountry.length ||
+    this.filter.selectedFlag.length ||
+    this.filter.selectedAttendance.length ||
+    this.filter.selectedScop
   )
     return true
   else return false
 }
 
-async function mounted (){
-  this.meetingsOptions = await getMajorMinorMeetings(this)
+async function onOrganizationSearch (query){
+  this.isLoadingOrganization = true
+  if (query)
+    this.organizationOptions = await this.$kronosApi.getOrganizations({
+      FreeText: query,
+      limit   : 25
+    })
 
-  this.organizationOptions = await this.$kronosApi.getOrganizations({
-    limit: 5000
-  })
+  this.isLoadingOrganization = false
+}
 
-  const country = await this.$kronosApi.getCountries()
+async function setQueryStringFilters (qs){
+  if (!_.isEmpty(qs)){
+    if (qs.FreeText) this.searchTagInput = qs.FreeText
+    selectedCountries.call(this, qs.Governments)
+    selectedMeetings.call(this, qs.EventUIDs)
+    selectCountryScope.call(this, qs.CountryScope)
+    selectedRegistrationStatus.call(this, qs.EventRegistrationStatus)
+    await selectedOrganizations.call(this, qs.OrganizationUIDs)
+  }
+}
+
+async function selectedOrganizations (organizationUIDs){
+  if (organizationUIDs){
+    this.organizationOptions = await this.$kronosApi.getOrganizations({
+      OrganizationUIDs: asArray(organizationUIDs)
+    })
+    const uids = asArray(organizationUIDs)
 
-  sortCountryByAsc(country, this)
+    this.filter.selectedOrganizations = this.organizationOptions.filter(o =>
+      uids.includes(o.OrganizationUID)
+    )
+  }
+}
 
-  this.organizationTypesOptions = await this.$kronosApi.getOrganizationTypes()
+function selectedRegistrationStatus (regStatus){
+  if (regStatus)
+    this.filter.selectedAttendance = asArray(regStatus).map(a =>
+      this.attendanceOptions.find(x => x.value === Number(a))
+    )
```

Comment:
this map(find) pattern is not confusing. filter/includes is more natural

eg:
```suggestion

     let statuses = asArray(regStatus);
     this.filter.selectedAttendance = this.attendanceOptions.filter(a => statuses.includes(a))
```